policy/lib/rule_data.rego

```diff
@@ -0,0 +1,27 @@
+package lib
+
+#
+# Will define built-in default values here.
+# They can be overridden if required.
+#
+# _test_key_3 is for the tests only. Not sure how else to get to 100% coverage.
+#
+rule_data_defaults := {"_test_key_3": 10}
+
+#
+# Returns the "first found" of the following:
+#   data.rule_data_custom[key_name]
+#   data.rule_data[key_name]
+#   rule_data_defaults[key_name]
+#
+# And falls back to an empty list if the key is not found anywhere.
+#
+rule_data(key_name) := value {
+	value := data.rule_data_custom[key_name]
+} else := value {
+	value := data.rule_data[key_name]
+} else := value {
+	value := rule_data_defaults[key_name]
+} else := value {
+	value := []
```

Comment:
I wonder if there's an idiomatic good way to throw an error here.

Returning a null would be another alternative.

Reply:
Not sure what happens if I just removed line 25 and 26 entirely, but I'm worried the answer is rules start being silently skipped.

Reply:
I think you're right because that would mean rule_data produces no result.